Notify/Notify.Functions/Notify.Functions/NotifyFunctions/Notification/UpdateNotification.cs

```diff
@@ -0,0 +1,108 @@
+        {
+            dynamic data = await ConversionUtils.ExtractBodyContentAsync(req);
+            ActionResult result;
+
+            log.LogInformation($"Got client's HTTP request to update notification based on {type}");
+            
+            try
+            {
+                result = await updateNotificationAsync(data, log, type);
+            }
+            catch (Exception ex)
+            {
+                log.LogError(ex, $"Error updating notification. Error message: {ex.Message}");
+                result = new BadRequestObjectResult(ex.Message);
+            }
+            
+            return result;
+        }
+
+        private static async Task<ActionResult> updateNotificationAsync(dynamic data, ILogger log, string type)
+        {
+            string lowerCasedType = type.ToLower();
+            IMongoCollection<BsonDocument> collection = MongoUtils.GetCollection(Constants.COLLECTION_NOTIFICATION);
+            FilterDefinition<BsonDocument> filter = Builders<BsonDocument>.Filter.Eq("_id", ObjectId.Parse(Convert.ToString(data.id)));
+            BsonDocument notificationDocument = await collection.FindAsync(filter).Result.FirstOrDefaultAsync();
+            UpdateDefinitionBuilder<BsonDocument> updateBuilder = Builders<BsonDocument>.Update;
+            List<UpdateDefinition<BsonDocument>> updates = new List<UpdateDefinition<BsonDocument>>();
+            
+            checkIfNotificationsCorrelates(data, lowerCasedType, notificationDocument);
+            
+            updates.Add(updateBuilder.Set("notification.name", Convert.ToString(data.name)));
+            updates.Add(updateBuilder.Set("description", Convert.ToString(data.description)));
+
+            if (lowerCasedType.Equals("location"))
+            {
+                updates.Add(updateBuilder.Set("notification.location", Convert.ToString(data.location)));
+                updates.Add(updateBuilder.Set("notification.permanent", Convert.ToString(data.permanent)));
+                updates.Add(updateBuilder.Set("notification.location", Convert.ToString(data.location)));
+            }
+            else if (lowerCasedType.Equals("dynamic"))
+            {
+                updates.Add(updateBuilder.Set("notification.location", Convert.ToString(data.location)));
+            }
+            else if (lowerCasedType.Equals("time"))
+            {
+                updates.Add(updateBuilder.Set("notification.timestamp", Convert.ToInt64(data.timestamp)));
+            }
+            else
+            {
+                throw new ArgumentException($"Type {type} is not supported");
+            }
+
+            log.LogInformation($"Attempting to update notification {data.id}");
+            await collection.UpdateOneAsync(filter, updateBuilder.Combine(updates));
+            log.LogInformation($"Successfully updated notification {data.id}");
+            
+            return new OkObjectResult($"Successfully updated notification {data.id}");
```

Comment:
Is it possible to use result here?

Reply:
This is the return value from the updateNotificationAsync method, and this is the value of the "result" variable on the main method of the function.